Use schema from enum wrapped in array

bagbag opened the pull request:
When enum arrays are used, the generated migrations (and likely other things) ignore the specified schema. This change checks for enums wrapped in arrays, in addition to non-array enums.

Should fix #1564, #3278, #3514 (and possibly other I havn't found).

devdatkumar commented on drizzle-kit/src/serializer/pgSerializer.ts:
```diff
@@ -158,7 +159,12 @@ export const generatePgSnapshot = (
 
-			const typeSchema = is(column, PgEnumColumn) ? column.enum.schema || 'public' : undefined;
+			const typeSchema = is(column, PgEnumColumn)
+				? column.enum.schema || 'public'
```
for line 163 and 165, use ?? instead of || to avoid falling back on 'public' if the schema is an empty string ''.
```tsx
const typeSchema = is(column, PgEnumColumn)
  ? column.enum.schema ?? 'public'
  : is(column, PgArray) && is(column.baseColumn, PgEnumColumn)
    ? column.baseColumn.enum.schema ?? 'public'
    : undefined;

```

devdatkumar commented on drizzle-kit/src/serializer/pgSerializer.ts:
```diff
@@ -746,7 +752,12 @@ export const generatePgSnapshot = (
+					: is(column, PgArray) && is(column.baseColumn, PgEnumColumn)
+						? column.baseColumn.enum.schema || 'public'
+						: undefined;
+
```
for line 756 and 758, use ?? instead of || to avoid falling back on 'public' if the schema is an empty string ''.
```tsx
const typeSchema = is(column, PgEnumColumn)
  ? column.enum.schema ?? 'public'
  : is(column, PgArray) && is(column.baseColumn, PgEnumColumn)
    ? column.baseColumn.enum.schema ?? 'public'
    : undefined;

```

devdatkumar reviewed (changes requested):
for line 163,165, 756, and 758, use ?? instead of || to avoid falling back on 'public' if the schema is an empty string ''.
```tsx
const typeSchema = is(column, PgEnumColumn)
  ? column.enum.schema ?? 'public'
  : is(column, PgArray) && is(column.baseColumn, PgEnumColumn)
    ? column.baseColumn.enum.schema ?? 'public'
    : undefined;

```

devdatkumar commented:
@AndriiSherman please review this.

bagbag committed:
use getEnumSchema for selectedFields

bagbag commented:
I've seen there were some changes to this in a after my PR. I've rebased, changed `||` to `??` and added the change made outside this PR to lines ~750 too.

apancutt commented:
I've just hit a semi-related issue which would be great if the fix could be included in this PR: drizzle-kit is not including the schema name when native TS enums are used. This tweak fixes it locally:

```diff
- return is(column, PgEnumColumn) ? column.enum.schema ?? 'public' : undefined;
+ return (is(column, PgEnumColumn) || is(column, PgEnumObjectColumn)) ? column.enum.schema ?? 'public' : undefined;
```
